src/client/Client.cc

```diff
@@ -7041,6 +7037,34 @@ int Client::_do_setattr(Inode *in, struct ceph_statx *stx, int mask,
     req->inode_drop |= CEPH_CAP_FILE_CACHE | CEPH_CAP_FILE_RD |
       CEPH_CAP_FILE_WR;
   }
+
+  if (mask & CEPH_SETATTR_MODE) {
+    req->head.args.setattr.mode = stx->stx_mode;
+    req->inode_drop |= CEPH_CAP_AUTH_SHARED;
+     
+    if (in->worm.is_enable() && !in->worm.is_retain() && !(stx->stx_mode & 00222)) {
```

Comment:
I'm not sure I really like treating a mode with no write bits as special.

What about POSIX ACLs here? Suppose there are no write bits in the mode but there is a POSIX ACL set that explicitly allows write access for another user? Should that trigger this transition?

A better mechanism might be to wire up the appropriate ioctl so that you could use "chattr -i" here. That would make it a privileged operation, but that seems reasonable for this.

Reply:
Thanks for your suggestion, jtlaytion!

My original intention was to find a simple way to trigger the conversion of the state and removing the Write property in mode was just one of the methods, and I didn't take into account the POSIX ACL scene. 

 I think your suggestion is more reasonable, but I'm not familiar with chattr,  so I think  I need a little time to learn how to do.

Reply:
Basically, you'd just need to wire up FS_IOC_GETFLAGS and FS_IOC_SETFLAGS and fix them up to handle the IMMUTABLE flag. FUSE has an ioctl handler, so it should be possible to do this for both the kernel and fuse.

Reply:
I'm very sorry for delay in reply.  I  add ll_set_flags() function to handle IMMUTABLE flag (make 
 the worm state transition in it) and ll_get_flags() function to get  the flags . Can you take a look at it?